lib/matplotlib/tests/test_table.py

```diff
@@ -229,3 +233,30 @@ def test_table_bbox(fig_test, fig_ref):
                   loc='center',
                   bbox=Bbox.from_extents(0.1, 0.2, 0.9, 0.8)
                   )
+
+
+@check_figures_equal(extensions=['.png'])
+def test_table_unit(fig_test, fig_ref):
+    # test that table doesn't participate in unit machinery, instead uses repr/str
+
+    class FakeUnit:
+        def __init__(self, thing):
+            pass
+        def __repr__(self):
+            return "Hello"
+
+    fake_convertor = munits.ConversionInterface()
+    fake_convertor.convert = Mock(side_effect=lambda v, u, a: 0)
+    # not used, here for completeness
+    fake_convertor.default_units = Mock(side_effect=lambda v, a: None)
+    fake_convertor.axisinfo = Mock(side_effect=lambda u, a: munits.AxisInfo())
+
+    munits.registry[FakeUnit] = fake_convertor
```

Comment:
It should never match anything else, but should this be cleaned out of the registry at the end of the test?